### PR TITLE
Catalog Update without having to hit bottom of page

### PR DESCRIPTION
**Pull request by JenniWhitman**
### What are the relevant tickets?
fixes https://github.com/mitodl/hq/issues/3141
fixes https://github.com/mitodl/hq/issues/2992
fixes https://github.com/mitodl/hq/issues/2910
fixes https://github.com/mitodl/hq/issues/2909

### Description (What does it do?)
Rather than relying on the page scroll to load more, there is a chain of requests to fill out the data for the page until it is full, regardless of page movement, once the initial load is complete.

### How can this be tested?
Set up data with more than 12 courses and/or programs (ideally both). You will want a few departments loaded as well.  Load the catalog and click to filter the list.  You should get all cards loading in.

### CODE REVIEW

File: frontend/public/src/containers/pages/CatalogPage.js [hunk 1]
```diff
@@ -71,116 +77,55 @@ export class CatalogPage extends React.Component<Props> {
...
-      )
-      this.setState({ filteredCourses: filteredCourses })
-
-      // Detect when the bottom of the catalog page has been reached and display more catalog items.
-      this.io = new window.IntersectionObserver(
-        this.bottomOfLoadedCatalogCallback,
-        { threshold: 1.0 }
-      )
-      this.io.observe(this.container.current)
+    console.log("gets here")
```
Comment (annagav) [inline frontend/public/src/containers/pages/CatalogPage.js:109]: let's remove this line

File: frontend/public/src/containers/pages/CatalogPage.js [hunk 2]
```diff
@@ -71,116 +77,55 @@ export class CatalogPage extends React.Component<Props> {
...
+    if (!coursesIsLoading) {
+      if (this.props.courses.length < this.props.coursesCount) {
+        const response = await getNextCoursePage(this.state.courseQueryPage + 1)
+        console.log(response)
```
Comment (annagav) [inline frontend/public/src/containers/pages/CatalogPage.js:113]: and this one

File: frontend/public/src/containers/pages/CatalogPage.js [hunk 3]
```diff
@@ -464,6 +409,10 @@ export class CatalogPage extends React.Component<Props> {
     )
   }
 
+  async loadMorePrograms() {
```
Comment (annagav) [inline frontend/public/src/containers/pages/CatalogPage.js:412]: is this needed?

**Review by annagav — CHANGES_REQUESTED** [review]
Few minor comments and failing js tests.

**Comment by annagav** [comment]
It says the page is unresponsive.
Screenshot 2024-02-13 at 9 02 17 AM

**Comment by JenniWhitman** [comment]
Closing as this was fixed in https://github.com/mitodl/mitxonline/pull/2130

**Pull request closed without merging**